planner/core/logical_plan_builder.go

```diff
@@ -300,8 +310,10 @@ func (b *PlanBuilder) buildJoin(joinNode *ast.Join) (LogicalPlan, error) {
 	switch joinNode.Tp {
 	case ast.LeftJoin:
 		joinPlan.JoinType = LeftOuterJoin
+		resetNotNullFlag(joinPlan.schema, leftPlan.Schema().Len(), joinPlan.schema.Len())
 	case ast.RightJoin:
 		joinPlan.JoinType = RightOuterJoin
+		resetNotNullFlag(joinPlan.schema, 0, leftPlan.Schema().Len()-1)
```

Comment:
The range of column indexes is `[start, end)`, I think the parameter for `end` in this function call should be `leftPlan.Schema().Len()`, not `leftPlan.Schema().Len()-1`.